### DISCUSSION TITLE
Null Exporter?

### DISCUSSION THREAD

**Discussion by jvilhuber** [open-telemetry/opentelemetry-collector-contrib#6797] [Ideas]
I'm wondering if there's a way to configure 'null' exporting, i.e. in some case (dev environments of some kind?) I may not want (or have) a place to send traces. Instead of reconfiguring the whole system (various gateways and components, and istio, etc) to not send traces, it seems useful if I leave the whole tracing plumbing in place to have all the various components still send traces to otel-collector, but otel-collector just silently discards them.
Other than writing a "null exporter", is there another way to accomplish this? I tried just configuring no exporters, but the collector fails to start with an error ("Error: invalid configuration: no enabled exporters specified in config").
Thoughts?

**Comment by jvilhuber** [comment]
During some other research I discovered this file: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/probabilisticsamplerprocessor/testdata/config.yaml
which seems to have 'nop' receiver/exporter/etc. Perhaps this is my answer?

**Reply by jpkrohling** [reply]
I don't think we add this to the actual distribution. I agree that this might be a useful exporter to have in contrib, for dev purposes. I'm moving this issue there.

**Reply by yurishkuro** [reply]
should convert to issue (or close if done)

**Reply by jpkrohling** [reply]
I'm closing, as I think the debug exporter could fill this role. The side effect is that it does minimal logging whenever it receives data, but I think it's a good compromise.

**Reply by Kampe** [reply]
I don't think this resolves this, why not allow a verbosity: null in debug?

**Reply by jpkrohling** [reply]
@Kampe, can you open this as an issue, if you haven't yet? I think a `verbosity: null` (or `none`) is actually a great idea.

**Reply by adeturner** [reply]
A null exporter does feel cleanest. My simple solution was to not create the exporter if I've got TRACER_NULL set.
